src/deferred.js

```diff
@@ -103,7 +103,10 @@ jQuery.extend( {
 						fns = null;
 					} ).promise();
 				},
-				then: function( onFulfilled, onRejected, onProgress ) {
+
+                // The onException is there to manage that an exception is thrown
+                // inside the onFullfilled without being catched.
+				then: function( onFulfilled, onRejected, onProgress, onException ) {
```

Comment:
We definitely don't want to expose new public functionality on Deferred without thinking carefully how it affects things. I think this should be a private API for now. It may be that we need some other implementation. I wonder if we could always use the [no-deferreds](https://github.com/jquery/jquery/blob/25d8ccd1112d75394b91071ff7eba13283aaf898/src/core/ready-no-deferred.js) implementation, it adds some bulk but that would need to be compared against the size increase here as well.